velox/functions/prestosql/aggregates/tests/ChecksumAggregateTest.cpp

```diff
@@ -0,0 +1,286 @@
+      O({1, 2}),
+      O({3, 4}),
+  });
+
+  assertChecksum(arrayVector, "/jjpuD6xkXs=");
+
+  arrayVector = makeVectorWithNullArrays<int64_t>({O({12, std::nullopt})});
+  assertChecksum(arrayVector, "sr3HNuzc+7Y=");
+  arrayVector = makeVectorWithNullArrays<int64_t>({O({1, 2}), std::nullopt});
+  assertChecksum(arrayVector, "Nlzernkj88A=");
+  arrayVector =
+      makeVectorWithNullArrays<int64_t>({O({1, 2}), std::nullopt, O({})});
+  assertChecksum(arrayVector, "Nlzernkj88A=");
+
+  // array of arrays
+  auto baseArrayVector = makeVectorWithNullArrays<int64_t>(
+      {O({1, 2}), O({3, 4}), O({4, std::nullopt}), O({})});
+  auto arrayOfArrayVector = makeArrayVector({0, 2}, baseArrayVector);
+  assertChecksum(arrayOfArrayVector, "Wp67EOfWZPA=");
+}
+
+TEST_F(ChecksumAggregateTest, maps) {
+  using p = std::pair<int64_t, double>;
+
+  auto mapVector = makeMapVector<int64_t, double>(
+      {{p(1, 17.0),
+        p(2, 36.0),
+        p(3, 8.0),
+        p(4, 28.0),
+        p(5, 24.0),
+        p(6, 32.0)}});
+
+  assertChecksum(mapVector, "T9pb6QUB4xM=");
+
+  auto mapOfArrays = createMapOfArraysVector<int64_t>(
+      {{{1, O({1, 2, 3})}}, {{2, O({4, 5, 6})}}, {{3, O({7, 8, 9})}}});
+
+  assertChecksum(mapOfArrays, "GGEqhJQZMa4=");
+
+  // map with nulls
+  auto mapWithNullArrays = createMapOfArraysVector<int64_t>(
+      {{{1, std::nullopt}},
+       {{2, O({4, 5, std::nullopt})}},
+       {{3, O({7, 8, 9})}}});
+
+  assertChecksum(mapWithNullArrays, "gwfQ1dI2P68=");
+}
+
+TEST_F(ChecksumAggregateTest, rows) {
+  auto row = makeRowVector(
+      {makeFlatVector<int64_t>({1, 3}), makeFlatVector<int64_t>({2, 4})});
+
+  assertChecksum(row, "jMIvLQ5YEVg=");
+
+  row = makeRowVector(
+      {makeNullableFlatVector<int64_t>({1, std::nullopt}),
+       makeNullableFlatVector<int64_t>({std::nullopt, 4})});
+
+  assertChecksum(row, "6jtxEIUj7Hg=");
+}
+
+TEST_F(ChecksumAggregateTest, globalAggregationNoData) {
+  auto row = std::vector{makeRowVector({makeFlatVector<int64_t>(0)})};
+  auto agg = PlanBuilder()
+                 .values(row)
+                 .singleAggregation({}, {"checksum(c0)"})
+                 .planNode();
+  assertQuery(agg, "VALUES (CAST(NULL AS VARCHAR))");
+
+  agg = PlanBuilder()
+            .values(row)
+            .partialAggregation({}, {"checksum(c0)"})
+            .intermediateAggregation()
+            .finalAggregation()
+            .planNode();
+  assertQuery(agg, "VALUES (CAST(NULL AS VARCHAR))");
+}
+
+TEST_F(ChecksumAggregateTest, timestampWithTimezone) {
+  auto timestamp = makeFlatVector<int64_t>(5, [](auto row) { return row; });
+  auto timezone = makeFlatVector<int16_t>(5, [](auto row) { return row % 5; });
+
+  auto timestampWithTzVector = std::make_shared<RowVector>(
+      pool_.get(),
+      TIMESTAMP_WITH_TIME_ZONE(),
+      BufferPtr(nullptr),
+      5,
+      std::vector<VectorPtr>{timestamp, timezone});
+
+  assertChecksum(timestampWithTzVector, "411aYzf4Rog=");
```

Comment:
You mentioned that checksum calculation for Timestamp with time zone is wrong. If this tests is not failing, then it means the expected value of the checksum here is wrong. Am I missing something?